### DISCUSSION TITLE
terminusdb CMS - base terminusdb.com on a terminusdb database installation

### DISCUSSION THREAD

**Discussion by annainfo** [terminusdb/terminusdb#1659] [Ideas]
Store all content of the terminusdb.com website in a terminusdb database and create any missing software to create a functioning website from that.

By doing so, demonstrate the confidence of the terminusdb organisation in using the software for their own business.

https://terminusdb.com/blog/terminusdb-a-headless-cms/
"Are we a WordPress or Drupal alternative? Certainly not. They serve a purpose for brochure websites, but not what we’re focused on. " 

WordPress is widely used, differences between installations can be described using a graph. For terminusdb.com only a subset of the features of WordPress is required.

**Comment by luke-feeney** [comment]
This is very much within scope right now - we are working on dogfooding the documentation first and will then do the full website. Really important that we demonstrate that confidence especially as we launch our CMS/KMS.

**Reply by annainfo** [reply]
Is this still worked on? Saw the change for the documentation, much appreciate that this was done, but the rest?

**Reply by alexander-mart** [reply]
@annainfo Dogfooding for site is good idea. But we already meet [problem with documentation](https://github.com/terminusdb/terminusdb/issues/2157). We need missing public collaboration tool based on TerminusDB CMS, otherwise we'll have to go back to good old git to track the history of edits to such an important thing as documentation in open source project.

**Reply by annainfo** [reply]
Re "otherwise we'll have to go back to good old git to track the history of edits" - texts from 2019 and 2022 about TerminusDB - "git for data (graph)"
1. 2019-11-05 Cheuk https://web.archive.org/web/20220527002338/https://terminusdb.com/blog/git-for-data/ "Last week I went to a meetup where a team of software developers from Dublin talked about their newest invention — TerminusDB, a graph database that stores data like git – A git for data graph!"
2. 2022-04-26 Gavin https://web.archive.org/web/20220427103234/https://terminusdb.com/blog/distributed-database-with-collaboration-model/ "TerminusDB is designed to be a distributed database with a collaboration model devised to rhyme with the ideas behind Git. It is essentially meant to be Git for data."

**Reply by hoijnet** [reply]
A new documentation site is now in place to serve the documentation needs until there is a full PR-capable public git-for-data TerminusDB capability up an running to host the documentation editing needs.

Dogfooding is important, and accurate documentation is more important right now. The new documentation site also has a new template assigned since today, contributions are more than welcome! 

https://terminusdb.org/docs

To place PRs for improvements (such as to fix the remaining table renderings), this is the location to push PRs to:

https://github.com/dfrnt-labs/terminusdb-docs-static

It reads the output from the respective client content in the original json file which are cached in the repo (javascript and python clients), and dynamic builds are not yet enabled in the build process, it is expected in the next iteration and is easy since we already render from the json.

Key thing was to enable new documentation and documentation updates to flow as PRs from the community 🤩

**Comment by hoijnet** [comment]
I both agree and disagree with the comment from @annainfo . Dogfooding is important, however it needs to balance with the ability for the community to leverage industry-standard protocols (regular git) for contributions.

For info, https://dfrnt.com/blog has been built from a TerminusDB data product for years. It makes sense as we are an internal team contributing. Getting the internal and external PR support would be great using the existing change request tooling. The base git-for-data works today, the additional parts for PRs requires industrialization and productization of the change request add-on software.

If there is community interest in supporting the efforts to improve the change request software, it could be coordinated into the roadmap ahead.

Getting documentation improved has a priority right now over the dogfooding. The current terminusdb.org is built from a TerminusDB repo, the next version will be built from a github repo until further notice and until the community can stand behind and be ready to support a productized PR change request functionality.

If there is commercial interest in the change request functionality for TerminusDB, that is an important consideration that may affect the roadmap ahead. (DM or add to this thread)

**Reply by annainfo** [reply]
Instead of improving a TerminusDB-based documentation, users can not help to improve a React-based documentation.

Users that prefer graph over React/js, now may have less trust. That adds to the focus of TerminusDB on GraphQL while not moving forward with ISO GQL.

IIRC Gavin in one video said, he believes in graph and every data should be in a graph? And now, not even a documentation?

But sorry, I have no insight in the details, last time that I tried to do something in TerminusCMS I was unhappy with the inconsistency.

Mediawiki and Wikibase have their documentation and revisions self-hosted. Is TerminusDB on the road to be a replacement for Wikibase?
